### PR TITLE
Enable snapshot-testing in base class

### PR DESCRIPTION
**Pull request by NicklasWallgren**
Re-iteration of https://github.com/origin-energy/java-snapshot-testing/pull/153 with support for JUnit4 and Spock.

```
@ExtendWith({SnapshotExtension.class})
public class SnapshotTestBase {
  protected Expect expect;

  protected void snapshot(Object value) {
    expect.toMatchSnapshot(value);
  }
}
```

### CODE REVIEW

File: java-snapshot-testing-core/src/main/java/au/com/origin/snapshots/ReflectionUtils.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import lombok.experimental.UtilityClass;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+@UtilityClass
+public class ReflectionUtils {
```
Comment (NicklasWallgren) [inline java-snapshot-testing-core/src/main/java/au/com/origin/snapshots/ReflectionUtils.java:11]: Should we move this util to a separate package, and add `@Internal` annotation?

Comment (jackmatt2) [reply java-snapshot-testing-core/src/main/java/au/com/origin/snapshots/ReflectionUtils.java:11]: Yes - I think so.

**Comment by jackmatt2** [comment]
I think you need to run `./gradlew spotlessApply` to apply formatting

**Comment by NicklasWallgren** [comment]
> I think you need to run `./gradlew spotlessApply` to apply formatting

Done

**Review by jackmatt2 — APPROVED** [review]

**Pull request merged**